src/lib/builtins/array.rs

```diff
@@ -929,4 +961,27 @@ mod tests {
         let second_in_many = forward(&mut engine, "duplicates.includes('d')");
         assert_eq!(second_in_many, String::from("false"));
     }
+
+    #[test]
+    fn for_each() {
+        let realm = Realm::create();
+        let mut engine = Executor::new(realm);
+        let init = r#"
+        var a = [2, 3, 4, 5];
+        var sum = 0;
+        var indexSum = 0;
+        var listLengthSum = 0;
+        function callingCallback(item, index, list) {
```

Comment:
You could test using `this` inside the callback.

Reply:
Actually, come to think of it, I am not sure `this` has been implemented. However, you could add a _TODO_ for when it is.